### DISCUSSION TITLE
Is it possible to make an exact replica of a system design from GitHub?

### DISCUSSION THREAD

**Discussion by ruru-m07** [community/community#110791] [GitHub Education]
### Hi everyone,

# GitHub-like Application with GCS Object Versioning

I'm currently working on an application similar to GitHub, and I'm using GCS object versioning. I've made progress in handling different versions of files, but I still need more help for the next steps.

So far, I've completed some minor tasks. You can visit my website at [Cocola Tech](https://www.cocola.tech/new) and create a free account. Please note that this project is still under development, so some buttons are only for viewing purposes. 

After creating your account, navigate to the `/new` page to create a new repository. You'll then be redirected to `/:yourusername/:yourreponame`. Here, you'll find a button labeled 'add files'. Click on it to upload some files and folders. Make sure to include a `README.md` file. 

Once you've uploaded the files, create a commit. Then, go back to 'add files' and upload an updated `README.md` file. You'll be able to see the updated files on the repository page.

# login page
![image](https://github.com/community/community/assets/142723369/8924243d-f5d9-44f8-92a3-6253f4fb4a9b)

# `/new` page
![image](https://github.com/community/community/assets/142723369/1fc285d7-7dfe-4446-91de-10ba34626cf6)

# upload page
![image](https://github.com/community/community/assets/142723369/f6b18602-7436-441a-9086-2b4ede76a84b)

# `/:yourusername/:yourreponame` page
![image](https://github.com/community/community/assets/142723369/3f2c2285-99e6-4280-bb34-0920549e6386)

**Comment by samus-aran** [comment]
Hi @ruru-m07 , we’re glad you’re here 😀

Thanks for sharing your project with the community. Unfortunately, we currently do not allow self-promotion or advertising on the Community Discussions. We want to make sure there is space for users to ask questions without overwhelming them with other conversations. Thank you for helping us maintain a productive and tidy community for all our members.
